src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -171,11 +265,10 @@ def forward(
         )
         query_states, key_states, value_states = mixed_qkv[0], mixed_qkv[1], mixed_qkv[2]
 
+        query_states = query_states * self.scale
```

Comment:
@amyeroberts I found that in the original model the scale was applied to q before the dot product q*k [2], and it looks like that was the reason that the outputs of the new HF models [1] were not close enough to the original implementation, you asked in the other PR [1] why tolerance i used was large.

What should i do about this change, as it could affect all the models that are using `Blip2Attention` e.g. blip_2, instructblip share the same code base, should a add config param or so?

[1] https://github.com/huggingface/transformers/pull/25612/files#r1406487831 your
[2] https://github.com/salesforce/LAVIS/blob/ac8fc98c93c02e2dfb727e24a361c4c309c8dbbc/lavis/models/eva_vit.py#L128

Reply:
Hmmm, this is a tricky one. Changing would break backwards compatibility, however, I'd be in favour if updating as it's effectively fixing a bug: we want our model to be as close as possible to the original implementation. In this case, can you update the PR name to include a 🚨 prefix? This will enable us to quickly spot the breaking change when preparing the next release an warn users

Reply:
Does this affect the slow tests?

i.e. does `RUN_SLOW=yes pytest tests/models/blip_2/test_modeling_blip2.py` still pass?

Reply:
yes, slow modeling_blip2 tests pass

Reply:
Thanks for checking!